wolfcrypt/src/asn.c

```diff
@@ -13727,6 +13727,11 @@ int wc_ValidateDate(const byte* date, byte format, int dateType)
     (void)tmpTime;
 
     ltime = wc_Time(0);
+    if (ltime < 0){
```

Comment:
It appears `time_t` is typically signed and is a `long`. For 32-bit systems long is 32-bits. This check seems valid.